sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -410,6 +401,24 @@ object ColumnPruning extends Rule[LogicalPlan] {
 }
 
 /**
+ * Eliminate no-op Project and Window.
+ *
+ * Note: this rule should be executed just after ColumnPruning.
+ */
+object EliminateOperators extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transformUp {
```

Comment:
See the latest PR: https://github.com/apache/spark/pull/11745. It will enable it.